src/Cronos/CronParser.cs

```diff
@@ -203,29 +215,41 @@ private static unsafe void ParseEndOfString(ref char* pointer)
                         AcceptCharacter(ref pointer, 'K') &&
                         AcceptCharacter(ref pointer, 'L') &&
                         AcceptCharacter(ref pointer, 'Y'))
-                        return CronExpression.Weekly;
+                        return rng == null
+                            ? CronExpression.Weekly
+                            : CronExpression.WeeklyWithJitter(rng);
+
                     return null;
                 case 'Y':
                     if (AcceptCharacter(ref pointer, 'E') &&
                         AcceptCharacter(ref pointer, 'A') &&
                         AcceptCharacter(ref pointer, 'R') &&
                         AcceptCharacter(ref pointer, 'L') &&
                         AcceptCharacter(ref pointer, 'Y'))
-                        return CronExpression.Yearly;
+                        return rng == null
+                            ? CronExpression.Yearly
+                            : CronExpression.YearlyWithJitter(rng);
+
                     return null;
                 default:
                     pointer--;
                     return null;
             }
         }
 
-        private static unsafe ulong ParseField(CronField field, ref char* pointer, ref CronExpressionFlag flags)
+        private static unsafe ulong ParseField(CronField field, ref char* pointer, ref CronExpressionFlag flags, Random? rng)
         {
             if (Accept(ref pointer, '*') || Accept(ref pointer, '?'))
             {
                 if (field.CanDefineInterval) flags |= CronExpressionFlag.Interval;
                 return ParseStar(field, ref pointer);
             }
+            
+            if (Accept(ref pointer, 'H'))
+            {
+                if (field.CanDefineInterval) flags |= CronExpressionFlag.Interval;
```

Comment:
This line is interesting, and should be covered by a dedicated unit test, and the `Interval` flag relates to Daylight Saving Time transitions, please see https://github.com/HangfireIO/Cronos?tab=readme-ov-file#interval-expression.

The point is that the jitter feature is sensitive to DST transitions, since it can easily schedule occurrences to "dangerous" time like 01:00 and 02:00 AM. Since Cronos handles this automatically, no problems are expected, but we should ensure that both interval-based and non-interval-based expressions work as expected with `H` characters.